internal/internal_workflow_testsuite_test.go

```diff
@@ -1247,3 +1247,124 @@ func (s *WorkflowTestSuiteUnitTest) Test_WorkflowLocalActivityWithMockAndListene
 	s.NoError(err)
 	s.Equal("hello mock", result)
 }
+
+func (s *WorkflowTestSuiteUnitTest) Test_SignalChildWorkflow() {
+	// This test will send signal from parent to child, and then child will send back signal to ack. No mock is needed.
+	signalName := "test-signal-name"
+	signalData := "test-signal-data"
+	childWorkflowFn := func(ctx Context, parentExec WorkflowExecution) (string, error) {
+		var data string
+		GetSignalChannel(ctx, signalName).Receive(ctx, &data)
+
+		err := SignalExternalWorkflow(ctx, parentExec.ID, parentExec.RunID, signalName, data+"-received").Get(ctx, nil)
+		if err != nil {
+			return "", err
+		}
+
+		return data + "-processed", nil
+	}
+
+	workflowFn := func(ctx Context) error {
+		cwo := ChildWorkflowOptions{ExecutionStartToCloseTimeout: time.Minute}
+		ctx = WithChildWorkflowOptions(ctx, cwo)
+		childFuture := ExecuteChildWorkflow(ctx, childWorkflowFn, GetWorkflowInfo(ctx).WorkflowExecution)
+
+		// wait until child starts
+		var childWE WorkflowExecution
+		if err := childFuture.GetChildWorkflowExecution().Get(ctx, &childWE); err != nil {
+			return err
+		}
+
+		// send signal to child workflow
+		signalFuture := SignalExternalWorkflow(ctx, childWE.ID, childWE.RunID, signalName, signalData)
```

Comment:
Let's add SignalChildWorkflow method to the childFuture. Internally it would wait for child start.
You can make it a separate PR.

Reply:
yes, already plan to do so.